packages/addon/lua/gelly/mods/blood-mod/init.lua

```diff
@@ -96,21 +128,40 @@ end
 hook.Add(
 	"GellyXDamage",
 	"gelly.builtin.blood-mod",
-	function(_, attacker, position, force, damage, type)
+	function(victim, attacker, position, force, damage, type)
 		if
-			not attacker:IsPlayer()
-			and not attacker:IsNPC()
-			and attacker:GetClass() ~= "prop_ragdoll"
+			not victim:IsValid() or
+			not victim:IsPlayer() and
+			not victim:IsNPC() and
+			not victim:IsRagdoll()
 		then
 			return
 		end
-		-- 8194 is the combination used in M9K for bullets
-		if type == 8194 then
-			type = 2
+
+		local bloodcolor = 0
+
+		if victim:GetBloodColor() ~= nil then
+			bloodcolor = victim:GetBloodColor()
 		end
+		
+		local oldabsorption = gellyx.presets.getActivePreset().Material.Absorption
+		local material = gellyx.presets.getActivePreset().Material
+
+		if bloodcolor == 0 then -- the blood is red >>> ALSO WHEN IT'S RAGDOLL SINCE THEY DON'T HAVE BLOOD COLOR <<<
+			material.Absorption = Vector(0.05, 0.5, 0.5)
+		elseif bloodcolor == 2 or bloodcolor == 4 then  -- the blood is yellow
+			material.Absorption = Vector(0.25, 0.3, 0.5)
+		else 											-- the blood is green
+			material.Absorption = Vector(0.3, 0.25, 0.5)
+		end 
+
+		gelly.SetFluidMaterial(material)
```

Comment:
@yogwoggf is there a way to set a material that isn't like a C graphics API (e.g. can we pass it to `gellyx.emitters.Sphere`)?

Reply:
Technically yes, but there's no specific API for this now. The closest is the latest ephemeral presets which are in-memory presets you can select that change the material and fluid parameters.

I'll look into opening a new API to modify a preset for a specific emission

Reply:
Dont presets include physical properties? I thought flex only supported global particle properties

Reply:
Yes, which is exactly why I coupled presets to the material. Although this is not really preferable for some situations, which is why I am looking into a solution.

Reply:
But materials are purely our application code no? As in we could create multiple instances and tag each particle rather than being limited to just one

Reply:
What I’m saying is:
- Physical properties are set globally, so i dont understand how you can make them “ephemeral”
- Material properties can be per particle, so should not be coupled to physical properties **at all**. You should be able to pass a material when spawning particles and it tags those with the material